controlplane/kubeadm/controllers/controller.go

```diff
@@ -413,6 +428,41 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 	return nil
 }
 
+// ReconcileInfrastructureMachineRefs adds infrastructure template ref as annotation to the infrastructure objects missing it.
+func (r *KubeadmControlPlaneReconciler) ReconcileInfrastructureMachineRefs(ctx context.Context, infraTemplateRef corev1.ObjectReference, machines internal.FilterableMachineCollection) (bool, error) {
+	isPatched := false
+	for _, machine := range machines {
+
+		infraObj, err := external.Get(ctx, r.Client, &machine.Spec.InfrastructureRef, machine.Spec.InfrastructureRef.Namespace)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return isPatched, errors.Wrapf(err, "Machine's infrastructure object is missing.")
+			}
+			return isPatched, errors.Wrapf(err, "Failed to get Machine's infrastructure object.")
```

Comment:
```suggestion
			return isPatched, errors.Wrapf(err, "failed to get Machine's infrastructure object")
```

Should we continue instead of returning here?